uniswap-v2/logics/impls/router/router.rs

```diff
@@ -425,20 +426,22 @@ impl<T: Storage<data::Data>> Router for T {
         Ok(())
     }
 
-    /// Original Uniswap Library pairFor function calculate pair contract address without making cross contract calls.
-    /// Please refer https://github.com/Uniswap/v2-periphery/blob/master/contracts/libraries/UniswapV2Library.sol#L18
-    ///
-    /// In this contract, use cross contract call to get pair contract address.
     default fn _pair_for(
         &self,
         factory: AccountId,
         token_a: AccountId,
         token_b: AccountId,
     ) -> Result<AccountId, RouterError> {
-        let (token_0, token_1) = self._sort_tokens(token_a, token_b)?;
-        let pair =
-            FactoryRef::get_pair(&factory, token_0, token_1).ok_or(RouterError::PairNotFound)?;
-        Ok(pair)
+        // TODO: remove pair code hash after deployment
```

Comment:
Can you add at least one or two tests (in Rust)?

Reply:
It is quite hard to test it in rust for now. Ideally, we should verify the computed address is equal to queried via cross contract call; but currently we cannot do cross-contract call in rust side.
If I need to add a test, then it will be a integration test in typescript.

Reply:
I think passing existing tests is sufficient to exhibit it works fine.

Reply:
I meant a hardcoded test with Alice as deployer & pari_code hash & arbitrary salt - > and checks if is equal what you get on-chain.
But seems like the code on substrate get refactored so it can be impl as is in the contracts so there is not much risk here. So agree it can be done in TS.
However, would you agree that this generate_address should be par of ink_env crate directly? (PR idea)

Reply:
Yep, I felt also that generate_address should be available from inv_env.